src/azul/terra.py

```diff
@@ -456,10 +458,11 @@ def run_sql(self, query: str) -> BigQueryRows:
                 job: QueryJob = bigquery.query(query)
                 try:
                     result = job.result()
-                except Forbidden as e:
-                    if 'Exceeded rate limits' in e.message and delay is not None:
-                        log.warning('Exceeded BigQuery rate limit during attempt %i/%i. '
-                                    'Retrying in %is.',
+                except (Forbidden, InternalServerError, ServiceUnavailable) as e:
+                    if delay and (
+                        type(e).__name__ != 'Forbidden' or 'Exceeded rate limits' in e.message
```

Comment:
This is not how one typically checks if an object is an instance of a class. You'd have to use the pythonic way or explain in a comment why you can't.